pkg/piecestore/psserver/agreementsender/agreementsender.go

```diff
@@ -79,31 +85,37 @@ func (as *AgreementSender) SendAgreementsToSatellite(ctx context.Context, satID
 	}
 	client := pb.NewBandwidthClient(conn)
 	defer func() {
-		err := conn.Close()
-		if err != nil {
+		if err := conn.Close(); err != nil {
 			as.log.Warn("Agreementsender failed to close connection", zap.Error(err))
 		}
 	}()
 
 	//todo:  stop sending these one-by-one, send all at once
 	for _, agreement := range agreements {
 		rba := agreement.Agreement
+		// Send agreement to satellite
+		sat, err := client.BandwidthAgreements(ctx, &rba)
 		if err != nil {
-			as.log.Warn("Agreementsender failed to deserialize agreement : will delete", zap.Error(err))
+			switch sat.GetStatus() {
+			case pb.AgreementsSummary_FAIL:
+				// CASE FAILED: connection with sat couldnt be established or connection
+				// is established but lost before receiving response from satellite
+				// no updates to the bwa status is done, so it remains "UNSENT"
+				as.log.Warn("Agreementsender lost connection", zap.Error(err))
+			default:
```

Comment:
I wonder if it's better to set default to FAIL rather than REJECTED

Reply:
hmm not sure here. if you feel strong about it I can change it.

Reply:
No strong feelings. I'm fine with merging as is